app/src/androidTest/java/org/simple/clinic/MigrationAndroidTest.kt

```diff
@@ -391,5 +413,99 @@ class MigrationAndroidTest {
       assertThat(it.count).isEqualTo(2)
     }
   }
-}
 
+  @Test
+  fun migration_17_to_18() {
+    val db_v17 = helper.createDatabase(version = 17)
+
+    val nonExistentFacilityUuid = "non-existent-facility-uuid"
+    val existentFacilityUuid = "464bcda8-b26a-484d-bb70-49b3675f4a38"
+
+    val measurementUuid1 = "ee367a66-f47e-42d8-965b-7a2b5c54f4bd"
+    val prescribedDrugUuid1 = "814132f2-f440-40cb-991c-43861c295815"
+
+    val measurementUuid2 = "fe367a66-f47e-42d8-965b-7a2b5c54f4bd"
+    val prescribedDrugUuid2 = "124132f2-f440-40cb-991c-43861c295815"
+
+    val userUuid = "a1d33096-cea6-4beb-8441-82cab2befe2d"
+    val patientUuid = "464bcda8-b26a-484d-bb70-59b3675f4a38"
+
+    fun insertFacilityStatement(uuid: String, name: String) = """
+      INSERT INTO "Facility" VALUES (
+        '$uuid',
+        '$name',
+        'Facility type',
+        'Street address',
+        'Village or colony',
+        'District',
+        'State',
+        'Country',
+        'Pin code',
+        '2018-09-25T11:20:42.008Z',
+        '2018-09-25T11:20:42.008Z',
+        'PENDING')
+    """
+
+    fun insertBloodPressureStatement(uuid: String, facilityUuid: String) = """
+      INSERT INTO "BloodPressureMeasurement" VALUES(
+        '$uuid',
+        120,
+        110,
+        'IN_FLIGHT',
+        '$userUuid',
+        '$facilityUuid',
+        '$patientUuid',
+        '2018-09-25T11:20:42.008Z',
+        '2018-09-25T11:20:42.008Z')
+    """
+
+    fun insertPrescribedDrugStatement(uuid: String, facilityUuid: String) = """
+      INSERT INTO "PrescribedDrug" VALUES(
+        '$uuid',
+        'Drug name',
+        'Dosage',
+        'rxNormCode',
+        0,
+        1,
+        '$patientUuid',
+        '$facilityUuid',
+        'PENDING',
+        '2018-09-25T11:20:42.008Z',
+        '2018-09-25T11:20:42.008Z')
+    """
+
+    db_v17.execSQL(insertFacilityStatement(uuid = existentFacilityUuid, name = "Facility name"))
+    db_v17.execSQL(insertBloodPressureStatement(uuid = measurementUuid1, facilityUuid = existentFacilityUuid))
+    db_v17.execSQL(insertPrescribedDrugStatement(uuid = prescribedDrugUuid1, facilityUuid = existentFacilityUuid))
+
+    try {
+      db_v17.execSQL(insertBloodPressureStatement(uuid = measurementUuid2, facilityUuid = nonExistentFacilityUuid))
+      db_v17.execSQL(insertPrescribedDrugStatement(uuid = prescribedDrugUuid2, facilityUuid = nonExistentFacilityUuid))
+
+    } catch (e: Throwable) {
+      assertThat(e).isInstanceOf(SQLiteConstraintException::class.java)
```

Comment:
There is still a problem here where the test will still pass if the error does not get thrown in the first place. I'll fix this.